packages/deployments/contracts/contracts/nomad-xapps/contracts/relayer-fee-router/RelayerFeeRouter.sol

```diff
@@ -0,0 +1,190 @@
+import {Router} from "../Router.sol";
+import {XAppConnectionClient} from "../XAppConnectionClient.sol";
+import {RelayerFeeMessage} from "./RelayerFeeMessage.sol";
+import {IBridgeToken} from "../../interfaces/bridge/IBridgeToken.sol";
+
+// ============ External Imports ============
+import {Home} from "../../../nomad-core/contracts/Home.sol";
+import {Version0} from "../../../nomad-core/contracts/Version0.sol";
+import {TypedMemView} from "../../../nomad-core/libs/TypedMemView.sol";
+
+// import "../../../../lib/forge-std/src/console.sol";
+
+/**
+ * @title BridgeRouter
+ */
+contract RelayerFeeRouter is Version0, Router {
+  // ============ Libraries ============
+
+  using TypedMemView for bytes;
+  using TypedMemView for bytes29;
+  using RelayerFeeMessage for bytes29;
+
+  // ========== Custom Errors ===========
+
+  error RelayerFeeRouter__onlyConnext_notConnext();
+  error RelayerFeeRouter__send_claimEmpty();
+  error RelayerFeeRouter__send_recipientEmpty();
+
+  // ============ Public Storage ============
+
+  IConnext public connext;
+
+  // ============ Upgrade Gap ============
+
+  // gap for upgrade safety
+  uint256[49] private __GAP;
+
+  // ======== Events =========
+
+  /**
+   * @notice Emitted when a fees claim has been initialized in this domain
+   * @param domain The domain where to claim the fees
+   * @param recipient The address of the relayer
+   * @param amount The amount of fees to claim
+   * @param transactionIds A group of transaction ids to claim for fee bumps
+   * @param remote Remote RelayerFeeRouter address
+   * @param message The message sent to the destination domain
+   */
+  event Send(uint32 domain, address recipient, uint256 amount, bytes32[] transactionIds, bytes32 remote, bytes message);
+
+  /**
+   * @notice Emitted when the a fees claim message has arrived to this domain
+   * @param originAndNonce Domain where the transfer originated and the unique identifier
+   * for the message from origin to destination, combined in a single field ((origin << 32) & nonce)
+   * @param origin Domain where the transfer originated
+   * @param recipient The address of the relayer
+   * @param amount The amount of fees to claim
+   * @param transactionIds A group of transaction ids to claim for fee bumps
+   */
+  event Receive(
+    uint64 indexed originAndNonce,
+    uint32 indexed origin,
+    address indexed recipient,
+    uint256 amount,
+    bytes32[] transactionIds
+  );
+
+  /**
+   * @notice Emitted when a new Connext address is set
+   * @param connext The new connext address
+   */
+  event SetConnext(address indexed connext);
+
+  // ======== Receive =======
+  receive() external payable {}
+
+  fallback() external payable {}
+
+  // ============ Modifiers ============
+
+  /**
+   * @notice Restricts the caller to the local bridge router
+   */
+  modifier onlyConnext() {
+    if (msg.sender != address(connext)) revert RelayerFeeRouter__onlyConnext_notConnext();
+    _;
+  }
```

Comment:
Might be cool to inherit this `connext` property, modifier, initialize, etc from a shared `abstract` contract between `BridgeRouter` and `RelayerFeeRouter`

Reply:
`RouterCore` ?

Reply:
I make sense to have shared things on a `ConnextRouter`

Reply:
Maybe overkill, but we could also have `ConnextRouter` create/initialize both `RelayerRouter` and `PromiseRouter` as children... like a `RouterManager` lol

Reply:
No we shouldn't have them all spawn like that, they should be independent instances and only really need to know about the core `Connext` contract